control/controller.go

```diff
@@ -422,6 +421,13 @@ func (q *Query) Done() {
 
 		q.transitionTo(Finished)
 		q.c.finish(q)
+
+		// count query request
+		if q.err != nil {
+			q.c.countQueryRequest(q, labelRuntimeError)
```

Comment:
I don't think this will catch all runtime errors.  It will only catch those errors that occur during execution initialization.  Errors that occur afterwards are just passed as results to the client, here:
https://github.com/influxdata/flux/blob/master/execute/result.go#L68
These errors are only reported to the client when `TableIterator` finally gets them.

There can be multiple errors that occur during runtime, maybe we need to change the `Query` interface to reflect this.  For now you could rebase on my branch and pull the errors from `flux.Statistics.RuntimeErrors` for the metric.

Maybe we should talk about how to handle error reporting moving forward, there is a deceptive amount of surface area for our client API.

Reply:
Shouldn't `Query.Done()` be called once the results have been consumed by the client?

At that point the `err` field should be properly populated.

Reply:
That seems intuitive, but it's not currently the case.

We just pass the `Result` from the `exec.Query` to the `control.Query`'s output channel here:
https://github.com/influxdata/flux/blob/master/control/controller.go#L617
When the client receives it, they call `result.Tables().Do(..)` to get at the tables in the result.  `Do` will return an error code if there is an error during execution, like a non-existent field reference or something like that.

So there's really three different error paths:
- Compile time error: returned from call to `controller.Query(..)`
- Execution initialization time error: returned from `query.Err()`
- Run time error: returned from call to `result.Tables().Do(...)`

I don't think we should be reporting the same error in multiple places, so it seems unclear to me if we want to return errors produced by `Tables().Do(...)` in `query.Err()` as well.

For reference, my PR for logging runtime errors is here:
https://github.com/influxdata/flux/pull/1227
https://github.com/influxdata/idpe/pull/3336

Reply:
Thank you Chris, I took a look at your PRs.
I was forgetting the `r.Tables().Do()` path. 👍